### PR TITLE
Adding get_item_id functions for the 4 types of stories

### PR DESCRIPTION
**Pull request by smritigambhir**
get_item_ids_for_<xyz>_stories and corresponding get_<xyz>_stories have been added

### CODE REVIEW

**Comment by poplav** [comment]
I haven't had a lot of time to look into this, but it looks like Travis is failing.

File: R/hacker_news.r [hunk 1]
```diff
@@ -1,5 +1,21 @@
 base_url <- "https://hacker-news.firebaseio.com/v0"
 
+get_item_ids_for_top_stories<-function(filename="/topstories.json",limit=10){
+  fromJSON(paste(base_url, filename, sep=""))[1:limit]
+}
+
+get_item_ids_for_new_stories<-function(filename="/newstories.json",limit=10){
+  `fromJSON(paste(base_url, filename, sep=""))[1:limit]
```
Comment (poplav) [inline R/hacker_news.r]: Why the single quote?  I also see it in other `fromJSON` calls.

File: R/hacker_news.r [hunk 2]
```diff
@@ -12,15 +28,39 @@ base_url <- "https://hacker-news.firebaseio.com/v0"
...
+  })
+  process_stories(top_stories, pretty_print)
 }
 
+get_new_stories <- function(limit = 10, pretty_print = FALSE) {
+  new_stories <- lapply(get_item_ids_for_new_stories(), function(id) {
+    story_url <- paste(base_url, "/item/", id, ".json", sep="")
+    fromJSON(story_url)
+  })
+  process_stories(top_stories, pretty_print)
```
Comment (poplav) [inline R/hacker_news.r]: Shouldn't this be `new_stories`.  Same goes for the other calls to `process_stories` where they need to be updated from `top_stories`

**Comment by poplav** [comment]
Roxygen is also being used, so it would be nice if we added documentation as well but not needed.  You can read more about roxygen at https://logfc.wordpress.com/2014/02/24/roxygen2-documenting-r-functions/.  Documentation gets generated in the `man` directory when you are running `devtools::check()`  See the workflow section at http://r-pkgs.had.co.nz/check.html for more info.

**Comment by smritigambhir** [comment]
Thanks for pointing these out, @poplav .
I have fixed the issues you mentioned. Can add documentation with the next issue PR. I had one question, though. Why do we need separate methods for 4 stories? Is the JSON structure for their respective files expected to be different from each other, in future? If not, won't it be neater to have one common get_item_id_for_story(filename,limit) and one get_stories(type=new/top/ask/show,limit,pretty_print) for all 4 stories?

**Comment by poplav** [comment]
@slayah Yes, the approach of passing in type and not having 4 separate functions make sense.  This definitely makes sense for `get_item_id_for_story(filename, limit)`.  I think it also makes sense to have a `get_stories` that takes in a type parameter rather than those individual functions.

**Comment by smritigambhir** [comment]
Not sure why the Travis is failing. The function runs properly when I run it locally for all types. Here is an example : 

get_stories(type="new")
[[1]]
[[1]]$by
[1] "unitedacademics"

[[1]]$descendants
[1] 0

[[1]]$id
[1] 12737054

[[1]]$score
[1] 1

[[1]]$time
[1] 1476813693

[[1]]$title
[1] "Xenotransplants: How a Pig Could Save Your Life"

[[1]]$type
[1] "story"

[[1]]$url
[1] "http://www.ua-magazine.com/xenotransplants-how-a-pig-could-save-your-life/"

[[2]]
[[2]]$by
[1] "vfXander"

[[2]]$descendants
[1] 0

[[2]]$id
[1] 12737040

[[2]]$score
[1] 1

[[2]]$time
[1] 1476813624

[[2]]$title
[1] "Magnet: Attraction on the Spot"

[[2]]$type
[1] "story"

[[2]]$url
[1] "http://www.tryMag.net"

[[3]]
[[3]]$by
[1] "sciadvance"

[[3]]$descendants
[1] 0

[[3]]$id
[1] 12737013

[[3]]$score
[1] 1

[[3]]$time
[1] 1476813406

[[3]]$title
[1] "Imaging advance offers new view of allergic asthma"

[[3]]$type
[1] "story"

[[3]]$url
[1] "https://directorsblog.nih.gov/2016/10/18/imaging-advance-offers-new-view-on-allergic-asthma/"

[[4]]
[[4]]$by
[1] "Brianjwma"

[[4]]$descendants
[1] 0

[[4]]$id
[1] 12737006

[[4]]$score
[1] 1

[[4]]$time
[1] 1476813371

[[4]]$title
[1] "This App Gets Freelancers Paid Faster"

[[4]]$type
[1] "story"

[[4]]$url
[1] "https://www.and.co/lp/general1"

[[5]]
[[5]]$by
[1] "nickfrost"

[[5]]$descendants
[1] 0

[[5]]$id
[1] 12736990

[[5]]$score
[1] 1

[[5]]$time
[1] 1476813246

[[5]]$title
[1] "Twitter Dips Below the Value of a Former Clone"

[[5]]$type
[1] "story"

[[5]]$url
[1] "https://mattermark.com/twitter-and/"

[[6]]
[[6]]$by
[1] "sama"

[[6]]$descendants
[1] 1

[[6]]$id
[1] 12736988

[[6]]$kids
[1] 12737035

[[6]]$score
[1] 2

[[6]]$time
[1] 1476813239

[[6]]$title
[1] "Why More Minority Founders Should Apply to Y Combinator"

[[6]]$type
[1] "story"

[[6]]$url
[1] "https://medium.com/@daveasalvant/why-more-minority-founders-should-apply-to-y-combinator-7e51208ea989#.mc8cd0qrq"

[[7]]
[[7]]$by
[1] "maxoliver"

[[7]]$descendants
[1] 0

[[7]]$id
[1] 12736979

[[7]]$score
[1] 1

[[7]]$time
[1] 1476813201

[[7]]$title
[1] "Interactive website showcasing importance of UX for startups and not only"

[[7]]$type
[1] "story"

[[7]]$url
[1] "http://ux.perfectial.com/?utm_source=social_blog&utm_medium=YCombinator&utm_campaign=WS16_tu1810"

[[8]]
[[8]]$by
[1] "puredanger"

[[8]]$descendants
[1] 0

[[8]]$id
[1] 12736965

[[8]]$score
[1] 1

[[8]]$time
[1] 1476813112

[[8]]$title
[1] "Updating Diagnostics for Boeing's 737 MAX [Clojure]"

[[8]]$type
[1] "story"

[[8]]$url
[1] "http://magazine.aerospacemanufacturinganddesign.com/article/october-2016/updating-diagnostics-for-boeings-737-max.aspx"

[[9]]
[[9]]$by
[1] "fictivmade"

[[9]]$descendants
[1] 0

[[9]]$id
[1] 12736964

[[9]]$score
[1] 2

[[9]]$time
[1] 1476813111

[[9]]$title
[1] "Designing for the Unknown: How NASA's JPL Engineers a Mission on Mars"

[[9]]$type
[1] "story"

[[9]]$url
[1] "http://blog.fictiv.com/posts/designing-for-the-unknown-how-nasas-jet-propulsion-lab-engineers-a-mission-on-mars"

[[10]]
[[10]]$by
[1] "atudoute"

[[10]]$descendants
[1] 0

[[10]]$id
[1] 12736956

[[10]]$score
[1] 1

[[10]]$time
[1] 1476813018

[[10]]$title
[1] "Nomad House: 32 customers, -$4500 from my personal accounts and 3 retreats later"

[[10]]$type
[1] "story"

[[10]]$url
[1] "https://medium.com/@atu/nomad-house-32-customers-4500-from-my-personal-accounts-and-3-retreats-later-d107ebbf9bff#---0-80.qdpccqnda"

File: R/hacker_news.r [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
 #' This function returns the top stories from hacker news
 #'
 #' @param limit  The top limit stories to return
 #' @param pretty_print  If true return list of serialized stories else return the json structure
 #' @export
 #' @examples
-#' get_top_stories()
-#' get_top_stories(limit = 5, pretty_print = TRUE)
+#' get_stories()
+#' get_stories(limit = 10, pretty_print = TRUE)
```
Comment (poplav) [inline R/hacker_news.r]: I think type is a required parameter.  In the build it runs these examples, so you may just need to add `type = "top"` here.

**Comment by poplav** [comment]
After the refactoring the tests still need to be updated see https://github.com/poplav/hackernews4r/blob/master/tests/testthat/test_get_top_stories.r#L4.  If those issues don't fix the build I'll take a look.  Thanks.

**Comment by smritigambhir** [comment]
Seems to have failed again after updating the test.

**Comment by smritigambhir** [comment]
Hey @poplav  Did you get a chance to take a look?

**Comment by poplav** [comment]
I will within the next 48 hours, sorry been a bit busy

**Comment by poplav** [comment]
@slayah.  I was looking into the travis failure (by running/testing locally) and the issue is:

```
cat /home/mike/prg/hackernews4r/hackernews4r.Rcheck/00install.out
* installing *source* package ‘hackernews4r’ ...
** R
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
Error in namespaceExport(ns, exports) : 
  undefined exports: get_top_stories
Error: loading failed
Execution halted
ERROR: loading failed
```

The issue was the package `NAMESPACE` was not properly updated.  The flow on how to use this is use https://github.com/hadley/devtools#package-development-tools.  Load devtools via `library(devtools)` at the root of your repo then `check()` check will generate proper namespaces and documentation based off the annotations in the code.

**Comment by smritigambhir** [comment]
Thanks @poplav  Yesterday itself I started looking into package structures in R and using roxygen. I actually was having trouble using install_github on my network here. It appears my DNS server is very slow and even using Google DNS server is not helping. So couldn't verify what was going wrong. I am actively looking into this though. Let me know if you have any suggestions.

**Comment by poplav** [comment]
@slayah If you saw I added a commit to get travis to pass, fixing the issues I noted in a previous comment.  I will merge this PR later today.  Thanks, again.

**Comment by smritigambhir** [comment]
@poplav Yes I saw that. Thanks for taking out the time :) Will try to have the PR for the second one ready once this gets merged.

**Comment by poplav** [comment]
Fixes #2

**Pull request merged**